packages/bot/src/scratch/index.js

```diff
@@ -81,6 +81,15 @@ export const runBot = (limitations = {}) => {
             block.setDisabled(true);
         }
     });
+
+    const { mandatoryMainBlocks } = config;
+    const has_main_blocks = mandatoryMainBlocks
```

Comment:
Better to rename this to `has_mandatory_blocks` for clarity because `has_main_blocks` implies the value is whether there are main blocks or not.

```suggestion
    const has_mandatory_blocks = mandatoryMainBlocks
```